src/starling/utils/RenderUtil.hx

```diff
@@ -235,7 +236,21 @@ class RenderUtil
         else if (#if (haxe_ver < 4.2) Std.is #else Std.isOfType #end(profile, String))
             profiles = [Std.string(profile)];
         else if (#if (haxe_ver < 4.2) Std.is #else Std.isOfType #end(profile, Array))
-            profiles = cast(profile, Array<Dynamic>);
+        {
+            var dynProfiles:Array<Dynamic> = cast(profile, Array<Dynamic>);
+            profiles = [];
+            for (prof in dynProfiles)
+            {
+                if (#if (haxe_ver < 4.2) Std.is #else Std.isOfType #end(prof, String))
+                {
+                    profiles.push(Std.string(prof));
+                }
+                else 
+                {
+                    profiles.push(prof);
```

Comment:
It's necessary to avoid the `Can't cast String to i32` error on HashLink target : my (poor) understanding is that if the value is typed as Dynamic, the Context3DProfile @:from isn't triggered for some reason

There is a similar assignment on lines (236-237), and those were already there (not sure why though)

I will limit that to HL target ( `#if hl ...` ) ?

Reply:
Does this work?

```haxe
for (prof in dynProfiles)
{
    profiles.push(Std.string(prof));
}
```

Reply:
It works if you pass an Array of String, not if you use Context3DProfile values because those are Ints

this will work : `_starling = new Starling(Game, stage, null, null, Context3DRenderMode.AUTO, ["baseline"]);`

and this won't : `_starling = new Starling(Game, stage, null, null, Context3DRenderMode.AUTO, [Context3DProfile.BASELINE]);`

Reply:
I updated my code with `#if hl` where needed + added a comment to remember why it's like that

I don't think I can update a PR ? I can close this one and create another

Reply:
also added code to handle a single Context3DProfile value -missed that possibility-
I realize that originally we are expected to pass Strings but it doesn't really make sense to me not to accept Context3DProfile values

Reply:
I hate that we use Ints for enums in OpenFL. Working on a better solution.

Reply:
This should work better. I feel that it's best to convert values to Context3DProfile as early as possible.

```haxe
var profiles:Array<Context3DProfile>;
var currentProfile:Context3DProfile;
```

```haxe
if (profile == "auto")
    profiles = ["enhanced",
                "standardExtended", "standard", "standardConstrained",
                "baselineExtended", "baseline", "baselineConstrained"];
else if (#if (haxe_ver < 4.2) Std.is #else Std.isOfType #end (profile, Int))
    profiles = [cast profile];
else if (#if (haxe_ver < 4.2) Std.is #else Std.isOfType #end (profile, String))
    profiles = [Std.string(profile)];
else if (#if (haxe_ver < 4.2) Std.is #else Std.isOfType #end (profile, Array))
{
    var dynProfiles:Array<Dynamic> = cast(profile, Array<Dynamic>);
    profiles = [];
    for (prof in dynProfiles)
    {
        if (#if (haxe_ver < 4.2) Std.is #else Std.isOfType #end (prof, Int))
            profiles.push(cast prof);
        else
            profiles.push(Std.string(prof));
    }
}
else
    throw new ArgumentError("Profile must be of type 'String' or 'Array'");
```

Reply:
> I don't think I can update a PR ? I can close this one and create another

Just commit/push to the same branch, and the PR will update automatically.

Reply:
> Just commit/push to the same branch, and the PR will update automatically.

Oh great ! I updated including your changes. I kept the `#if hl`  let me know if you prefer to remove them (I'd rather keep the comment explaining the` Std.string` thing though) ?